source/particle/world.cc

```diff
@@ -979,25 +979,33 @@ namespace aspect
 
       std::vector<Tensor<1,dim> >  result(particles_in_cell);
       std::vector<Tensor<1,dim> >  old_result(particles_in_cell);
-      std::vector<Point<dim> >     particle_points(particles_in_cell);
 
-      typename std::multimap<types::LevelInd, Particle<dim> >::iterator it = begin_particle;
-      for (unsigned int i = 0; it!=end_particle; ++it,++i)
-        {
-          particle_points[i] = it->second.get_reference_location();
-        }
+      std::vector<types::global_dof_index> cell_dof_indices (this->get_fe().dofs_per_cell);
 
-      const Quadrature<dim> quadrature_formula(particle_points);
-      FEValues<dim> fe_value (this->get_mapping(),
-                              this->get_fe(),
-                              quadrature_formula,
-                              update_values);
+      const FiniteElement<dim> &velocity_fe = this->get_fe().base_element(this->introspection().base_elements.velocities);
+      cell->get_dof_indices (cell_dof_indices);
 
-      fe_value.reinit (cell);
-      fe_value[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
-                                                                                 result);
-      fe_value[this->introspection().extractors.velocities].get_function_values (this->get_old_solution(),
-                                                                                 old_result);
+      for (unsigned int j=0; j<velocity_fe.dofs_per_cell; ++j)
+        {
+          Tensor<1,dim> solution;
+          Tensor<1,dim> old_solution;
```

Comment:
I had to twist my brain a bit to really understand what this code does. In essence, what these two variables do is to evaluate the solution *at a support point*. It is then later interpolated (via the shape functions). I think it may actually clearer if these variables were called `solution_at_support_point` or similar.